src/Mod/TechDraw/App/ProjectionAlgos.cpp

```diff
@@ -50,6 +50,18 @@ using namespace std;
 // ProjectionAlgos
 //===========================================================================
 
+namespace TechDraw {
+  //added by tanderson. aka blobfish.
+  //projection algorithms build a 2d curve(pcurve) but no 3d curve.
+  //this causes problems with meshing algorithms after save and load.
+  const TopoDS_Shape& build3dCurves(const TopoDS_Shape &shape)

```

Comment:
Did you mean to remove "static" on build3dCurves?   If it is static, it can be used without a ProjectionAlgo instance.

I don't think this routine is widely used without the rest of ProjectionAlgo, so maybe it doesn't matter.

Reply:
Yes, this was intentionally, thanks for noticing, because I realize that I should have mentioned my design choices. I can imagine you would like to see things a bit differently.

I believe `static` here does not mean that `build3dCurves()` is a class function (that doesn't require an instance). It's not part of the `ProjectionAlgo` class. Instead, I think the author of the code wanted to express that this is a small helper function local to this translation unit, so no other compiled needs to link to this function. However, for my changes, I do want to link to this function (from `AppTechDrawPy.cpp`), so I needed to remove the `static` keyword.

In addition, I needed to add the `TechDraw` namespace to make it accessible. However, I can imagine the location of this function is a bit strange now. It used to be a helper function local to the file for the `ProjectionAlgo` functions. Now it is a function accessible from the `TechDraw` namespace. I chose to leave the function in this file because it originates from here, for easier review, and because I couldn't find a better location, but I can imagine that you prefer to relocate the function, perhaps in a TechDraw file with similar kinds of utility functions.